R/mod_pkg_explorer.R

```diff
@@ -83,7 +85,17 @@ mod_pkg_explorer_server <- function(id, selected_pkg,
 
     nodes <- reactive({
       req(pkgdir())
-      make_nodes(list.files(pkgdir(), recursive = TRUE))
+      s <- make_nodes(list.files(pkgdir(), recursive = TRUE)) 
+      # attr(s[[1]][[1]],"stselected") = TRUE
+      # attr(s[[1]],"stopened") = TRUE
+      if(!is.null(s[["DESCRIPTION"]])){
+      attr(s[["DESCRIPTION"]],"stselected") = TRUE
+      }
+      else {
+        f <- names(head(map(s, \(x) attr(x,"sttype") == "file"),1))
```

Comment:
```suggestion
        f <- names(head(purrr::keep(s, \(x) !is.null(attr(x, "sttype"))), 1))
```
This doesn't always work. If the top line item is a folder, no files will be selected.

Reply:
corrected